kvbc/test/v4blockchain/v4_blockchain_test.cpp

```diff
@@ -740,6 +843,770 @@ TEST_F(v4_kvbc, delete_last_reachable) {
+      auto merkle_val = std::get<categorization::MerkleValue>(*val);
+      ASSERT_EQ(merkle_val.block_id, blk);
+      ASSERT_EQ(merkle_val.data, val_str);
+    }
+
+    for (uint32_t kid = 1; kid <= num_versioned_each; ++kid) {
+      std::string key = "versioned_key_" + std::to_string(blk) + "_" + std::to_string(kid);
+      std::string val_str = "versioned_value_" + std::to_string(blk) + "_" + std::to_string(kid);
+      versioned_keys.push_back(key);
+      versioned_versions.push_back(blk);
+      auto val = blockchain.get("versioned", key, blk);
+      ASSERT_TRUE(val.has_value());
+      auto ver_val = std::get<categorization::VersionedValue>(*val);
+      ASSERT_EQ(ver_val.block_id, blk);
+      ASSERT_EQ(ver_val.data, val_str);
+    }
+    for (uint32_t kid = 1; kid <= num_immutable_each; ++kid) {
+      std::string key = "immutable_key_" + std::to_string(blk) + "_" + std::to_string(kid);
+      std::string val_str = "immutable_value_" + std::to_string(blk) + "_" + std::to_string(kid);
+      immutable_keys.push_back(key);
+      immutable_versions.push_back(blk);
+      auto val = blockchain.get("immutable", key, blk);
+      ASSERT_TRUE(val.has_value());
+      auto immutable_val = std::get<categorization::ImmutableValue>(*val);
+      ASSERT_EQ(immutable_val.block_id, blk);
+      ASSERT_EQ(immutable_val.data, val_str);
+    }
+  }
+
+  ///////Checking the multiGet///////////////////////
+  blockchain.multiGet("merkle", merkle_keys, merkle_versions, values);
+  key_start = std::string("merkle_key_").size();
+  val_start = std::string("merkle_value_").size();
+  for (size_t i = 0; i < merkle_keys.size(); ++i) {
+    ASSERT_TRUE(values[i].has_value());
+    auto merkle_val = std::get<categorization::MerkleValue>(*(values[i]));
+    ASSERT_EQ(merkle_keys[i].substr(key_start), (merkle_val.data).substr(val_start));
+  }
+
+  blockchain.multiGet("versioned", versioned_keys, versioned_versions, values);
+  key_start = std::string("versioned_key_").size();
+  val_start = std::string("versioned_value_").size();
+  for (size_t i = 0; i < versioned_keys.size(); ++i) {
+    ASSERT_TRUE(values[i].has_value());
+    auto versioned_val = std::get<categorization::VersionedValue>(*(values[i]));
+    ASSERT_EQ(versioned_keys[i].substr(key_start), (versioned_val.data).substr(val_start));
+  }
+
+  blockchain.multiGet("immutable", immutable_keys, immutable_versions, values);
+  key_start = std::string("immutable_key_").size();
+  val_start = std::string("immutable_value_").size();
+  for (size_t i = 0; i < immutable_keys.size(); ++i) {
+    ASSERT_TRUE(values[i].has_value());
+    auto immutable_val = std::get<categorization::ImmutableValue>(*(values[i]));
+    ASSERT_EQ(immutable_keys[i].substr(key_start), (immutable_val.data).substr(val_start));
+  }
+
+  merkle_keys.clear();
+  merkle_versions.clear();
+  versioned_keys.clear();
+  versioned_versions.clear();
+  immutable_keys.clear();
+  immutable_versions.clear();
+  // Checking again
```

Comment:
you can add to the comment that the range includes the deletes, took me a minute to understad

Reply:
thanks @cloudnoize , i am adding the comment.